hs/CalendarResponse.hs

```diff
@@ -0,0 +1,281 @@
+            -> [[String]]
+eventsByCourse code session sect start end dates timeSystem =
+    if session == "Y" then year else halfCourse
+    where
+        year = halfFall ++ halfWinter
+        halfFall = eventsByTime code sect start end (dates !! 0) timeSystem
+        halfWinter = eventsByTime code sect start end (dates !! 1) timeSystem
+        halfCourse = eventsByTime code sect start end (dates !! 0) timeSystem
+
+-- | Creates an event for each start/end time and date.
+eventsByTime :: String -- ^ Course code.
+             -> String  -- ^ Course section.
+             -> [[String]] -- ^ Start times.
+             -> [[String]] -- ^ End times.
+             -> [(String, String)] -- ^ Start/End dates.
+             -> String -- ^ Current time.
+             -> [[String]]
+eventsByTime code sect start end dates timeSystem =
+    map (eventsByDate code sect timeSystem) (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' (x:xs) (y2:ys) (z:zs) = (x,y2,z):zip' xs ys zs
+zip' _ _ _ = []
+
+-- | Creates an event for each start/end time.
+eventsByDate :: String -- ^ Course code.
+             -> String -- ^ Course section.
+             -> String -- ^ Current time.
+             -> ([String], -- ^ Start time
+             [String], -- ^ End time
+             (String, String)) -- ^ Start/End date.
+             -> [String]
+eventsByDate code sect timeSystem (start, end, dates) =
+    (zip start end) >>= (eventsGenerator code sect timeSystem dates)
+
+-- | Generates the string that represents each event.
+eventsGenerator :: String -- ^ Course code.
+                -> String -- ^ Course section.
+                -> String -- ^ Current time.
+                -> (String, String) -- ^ Start/End date.
+                -> (String, String) -- ^ Start/End time.
+                -> [String]
+eventsGenerator code sect timeSystem (startDate, endDate) (start, end) =
+    if startDate == "" || endDate == ""
+    then []
+    else ["BEGIN:VEVENT", "DTSTAMP:" ++ timeSystem,
+          "DTSTART;TZID=America/Toronto:" ++ startDate ++ start,
+          "DTEND;TZID=America/Toronto:" ++ startDate ++ end,
+          "RRULE:FREQ=WEEKLY;UNTIL=" ++ endDate ++ "000000Z",
+          "ORGANIZER:University of Toronto",
+          "SUMMARY:" ++ code ++ " " ++ sect,
+          "CATEGORIES:EDUCATION", "END:VEVENT"]
+
+-- ** Ordering data
+
+-- | Orders the time fields for each course; which were obtained from the
+-- database
+orderData :: [Time] -> [[[Double]]]
+orderData courseFields = filter notNull (assignDay $ map timeField courseFields)
+    where
+        notNull field = not $ null field
+
+-- | Organizes the data by the day of the week, assigning a position based on
+-- the day in the time field.
+assignDay :: [[Double]] -> [[[Double]]]
+assignDay lst = [monday, tuesday, wednesday, thursday, friday]
+    where
+        monday = filter (\ field -> field !! 0 == 0.0) lst
```

Comment:
Some of the lambdas have spaces, some don't. Courseography generally does not have spaces between the lambda and the argument.